playbooks/tasks/portal-firewall-ufw-setup.yml

```diff
@@ -0,0 +1,61 @@
+---
+
+# Setup portal ufw firewall rules
+
+- block:
+    - name: Reset ufw (reset ufw rules and disables ufw firewall)
+      community.general.ufw:
+        state: reset
+    
+    - name: Limit SSH access
+      community.general.ufw:
+        rule: limit
+        direction: in
+        port: '22'
+        proto: tcp
+
+    - name: Allow outgoing traffic by default
+      community.general.ufw:
+        default: allow
+        direction: outgoing
+
+    - name: Deny incoming traffic by default
+      community.general.ufw:
+        default: deny
+        direction: incoming
+
+    - name: Allow specific incoming ports
```

Comment:
Follow Up: We should add comment explaining why each port is needed for reference in the future.